internal/pkg/controllers/profile/profile.go

```diff
@@ -204,15 +208,45 @@ func (r *Reconciler) reconcileSeccompProfile(
 		r.record.Event(sp, event.Warning(reasonCannotGetProfilePath, err))
 		return reconcile.Result{RequeueAfter: wait}, nil
 	}
+	nodeFinalizerString := os.Getenv(config.NodeNameEnvKey) + "-delete"
+	if !sp.ObjectMeta.DeletionTimestamp.IsZero() { // object is being deleted
```

Comment:
Do we still want to remove the profile if a workload references it? Restarting the workload would not work any more.

Reply:
What do you think the behavior should be? Should the `SeccompProfile` add an error condition but remain in an active state? Should it go into a terminating state until all workloads are deleted?

Reply:
Sticking in termination could be a good option. I’m not sure, WDYT?

@hasheddan @pjbgf any thoughts?

Reply:
I think the implementation for causing it to go into a terminating state to wait for all workloads to be deleted would involve getting every pod that references it to add a finalizer to it, which i think would be a little messy. I think it would be easier to simply fail to delete it and emit an event/condition saying why it failed to delete.

Reply:
Sounds good. :+1:

Reply:
In working through it I found that there wasn't really a way to just block deletion except with finalizers, so this latest update indexes pods that are using the profile and adds a finalizer until they are all removed.